Sources/Signal.swift

```diff
@@ -27,6 +27,12 @@ public final class Signal<Value, Error: Swift.Error> {
 	/// The state of the signal. `nil` if the signal has terminated.
 	private let state: Atomic<SignalState<Value, Error>?>
 
+	/// The status of the signal. It synchronizes with `sendLock`.
+	private var status: SignalStatus<Value, Error>
```

Comment:
I think having both `state` and `status` feels confusing. It's unclear to me whether they're two related concepts and should be together, or two independent concepts and perhaps could be named more descriptively?